dpcpp/components/format_conversion.dp.hpp

```diff
@@ -90,6 +92,43 @@ size_type calculate_nwarps(std::shared_ptr<const DpcppExecutor> exec,
 
 }  // namespace host_kernel
 }  // namespace coo
+
+
+namespace bccoo {
+namespace host_kernel {
+
+
+/**
+ * @internal
+ *
+ * It calculates the number of warps used in Coo Spmv depending on the GPU
+ * architecture and the number of stored elements.
+ */
+template <size_type subgroup_size = config::warp_size>
+size_type calculate_nwarps(std::shared_ptr<const DpcppExecutor> exec,
+                           const size_type nnz)
+{
+    size_type nwarps_in_dpcpp = exec->get_num_computing_units() * 7;
```

Comment:
```suggestion
    size_type nwarps_in_dpcpp =  exec->get_num_subgroups();
```
we introduce it recently to avoid hard-code 7

Reply:
I have to test it